tools/backups2datalad/asyncer.py

```diff
@@ -144,6 +146,23 @@ def log(self) -> PrefixedLogger:
     def repo(self) -> Path:
         return self.ds.pathobj
 
+    async def __aenter__(self) -> Downloader:
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[type[BaseException]],
+        _exc_val: Optional[BaseException],
+        _exc_tb: Optional[TracebackType],
+    ) -> None:
+        if exc_type is None:
+            if self.addurl is not None:
+                await self.addurl.aclose()
+        else:
+            with anyio.CancelScope(shield=True):
+                if self.addurl is not None:
+                    await self.addurl.force_aclose()
```

Comment:
may be smth like 
```suggestion
        with anyio.CancelScope(shield=True) if exc_type else contextlib.nullcontext():
            if self.addurl is not None:
                await self.addurl.force_aclose()
```
to avoid duplication?

Reply:
The two cases aren't the same; in the first, `aclose()` is called, but in the second, `force_aclose()` is.

Reply:
ah, missed that